tools/commands-packages-query.js

```diff
@@ -0,0 +1,1159 @@
+    return versionRecord;
+  },
+  // Print the query information to screen.
+  //
+  // options:
+  //   - ejson: Don't pretty-print the data. Return a machine-readable ejson
+  //     object.
+  print: function (options) {
+    var self = this;
+
+    // If we are asking for an EJSON-style output, we will only print out the
+    // relevant fields.
+    if (options.ejson) {
+      var versionFields = [
+        "name", "version", "description", "summary", "git", "dependencies",
+        "publishedBy", "publishedOn", "installed", "local", "OSarchitectures"
+      ];
+      var packageFields =
+        [ "name", "homepage", "maintainers", "versions", "totalVersions" ];
+      var fields = self.data.version ? versionFields : packageFields;
+      Console.rawInfo(convertToEJSON(_.pick(self.data, fields)));
+      return;
+    }
+
+    // Otherwise, display the information that we have. If we were asking about
+    // a specific version, display that. Otherwise, display package metadata in
+    // general.
+    if (self.data.version) {
+      self._displayVersion(self.data);
+      return;
+    }
+    self._displayPackage(self.data);
+  },
+  // Aggregates data about the package as a whole. Returns an object with the
+  // following keys:
+  //
+  // - name: package name
+  // - maintainers: an array of usernames of maintainers
+  // - homepage: string homepage
+  // - totalVersions: total number of versions that this package has, including
+  //   local and hidden versions.
+  // - defaultVersion: a default version: use this version to look up
+  //   per-version information that is relevant to the package as a whole, such
+  //   as git, description,etc.
+  // - versions: an array of objects representing versions of this package.
+  _collectPackageData: function () {
+    var self = this;
+    var data = {
+      name: self.metaRecord.name,
+      maintainers: _.pluck(self.metaRecord.maintainers, "username"),
+      homepage: self.metaRecord.homepage
+    };
+
+    // Collect surface information about available versions, starting with the
+    // versions available on the server.
+    var serverVersionRecords =
+          catalog.official.getSortedVersionRecords(self.name);
+    var totalVersions = serverVersionRecords.length;
+
+    // If we are not going to show hidden versions, then we shouldn't waste time
+    // on them. Trim the serverVersionRecords array to only have the top
+    // MAX_RECENT_VERSIONS migrated, official versions.
+    if (! self.showHiddenVersions) {
+      serverVersionRecords =
+        _.filter(serverVersionRecords, function (vr) {
+          return ! vr.unmigrated && vr.version.indexOf("-") === -1;
+      });
+      serverVersionRecords = _.last(serverVersionRecords, MAX_RECENT_VERSIONS);
+    };
+
+    // Process the catalog records into our preferred format, and look up any
+    // other per-version information that we might need.
+    data["versions"] = _.map(serverVersionRecords, function (versionRecord) {
```

Comment:
Hmm. This is something that we talked about when making the design doc, and decided on this ordering as being more intuitive! I am not super-attached to it either way, but I don't think that we should base this on Atmosphere. For example -- Atmosphere is in a browser, where you start at the top of the page. When (many people) run commands in (many common) terminals, the terminal scrolls to the bottom of the output by default. Hopefully the output of 'meteor show x' can fit in the terminal screen without scrolling, but 'meteor show --show-all x' often can't. It seems to make sense to me that, in that case, you should should be able to see the most recent versions first, and scroll up to see the older ones.